src/Controller/CRUDController.php

```diff
@@ -916,17 +903,15 @@ public function getRequest(): Request
     /**
      * Contextualize the admin class depends on the current request.
      *
-     * NEXT_MAJOR: Change \RuntimeException by \InvalidArgumentException in the next line.
-     *
      * @throws \RuntimeException
+     * @throws \InvalidArgumentException
      */
     final public function configureAdmin(Request $request): void
     {
         $adminCode = $request->get('_sonata_admin');
 
         if (null === $adminCode) {
-            // NEXT_MAJOR: Change \RuntimeException by \InvalidArgumentException in the next line.
-            throw new \RuntimeException(sprintf(
+            throw new \InvalidArgumentException(sprintf(
```

Comment:
Should this change be moved to a separate commit? It does not seem to fit the description.

Reply:
Makes sense

Reply:
👍 I will continue with this once `3.x` is merged in this branch and I'll split the PR.